flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/CallContext.java

```diff
@@ -93,6 +93,17 @@ default ValidationException newValidationError(String message, Object... args) {
         return new ValidationException(String.format(message, args));
     }
 
+    /**
+     * Shorthand for {@code if (throwOnFailure) throw ValidationException(...) else return
```

Comment:
The comment here is not useful, just copies the actual code.
And it's a utility method on an interface, can't we define it elsewhere as a static method?

Reply:
TBH I wasn't sure where to place it, I placed it here as it's related to `newValidationError`. TBH I would rather remove `newValidationError` at some point, in order to force users to always use this method instead.

Reply:
I get it, don't have a quick good alternative, just felt weird to have a "static" utility method in the iface, (same goes for the existing `newValidationError()`).